15-Appendix.Rmd

```diff
@@ -0,0 +1,144 @@
+| `install.packages('Rcpp')`        | `conda install r-Rcpp`        |
+
+![](images/conda/conda_install_rcpp.gif)
+
+##### Install a specific version of a package
+
+| `install.packages`                                                                                              | conda install                   |
+| --------------------------------------------------------------------------------------------------------------- | ------------------------------- |
+| `require(devtools)`<br> `install_version("ggplot2", version = "2.2.1", repos = "http://cran.us.r-project.org")` | `conda install r-ggplot2=2.2.1` |
+
+![](images/conda/conda_install_specific_version.gif)
+
+### Python
+
+Python packages do not require a prefix and can simply be installed using their
+name.
+
+#### Examples
+
+##### Install a package
+
+In the terminal run: `conda install numpy`
+
+which you can now access in your R session
+
+```r
+library(reticulate)
+np <- import("numpy")
+np$arange(15)
+```
+
+![](images/conda/np_from_conda.gif)
+
+### Operating System Packages
+
+Even if you want to continue using `packrat` to manage your R packages you can
+use conda to resolve operating system dependencies like libxml2
+
+#### Examples
+
+##### Installing a package that relies on OS dependency
+
+Suppose you want to install the R package `bigIntegerAlgos` but it fails because
+it depends on a system level library called `gmp`. Switch to the terminal and
+use `conda` to install it. Then switch back to the R console and try to use
+`install.packages` again
+
+![](images/conda/conda_install_with_os_dep_v2.gif)
+
+## Platform limitations
+
+Usually when using Conda, it makes sense to have one environment per project,
+but because we are using the Open Source version of R Studio, there is only a
+single Conda environment available.
+
+This means having to be careful to make sure packages don't pollute your
+environment from another project.
+
+## Environment Management
+
+### Reset your _Environment_ to default
+
+Recommended before starting a new project. Will ensure that no unused
+dependencies are exported when you export an `environment.yml` for this project.
+
+```bash
+conda env export -n base| grep -v "^prefix: " > /tmp/base.yml && conda env update --prune -n rstudio -f /tmp/base.yml && rm /tmp/base.yml
+```
+
+### Exporting your _Environment_
+
+This is similar to making a `packrat.lock` file, it catalogues all of the
+dependencies installed in your environment so that another user can restore a
+working environment for your application. Check this `environment.yml` file into
+your git repository.
+
+```bash
+conda env export | grep -v "^prefix: " > environment.yml
+```
+
+### Making your R Studio Environment match an `environment.yml`
+
+When checking out a project that has an `environment.yml` do this to install any
+packages required by the project that you don't have in your working
+environment.
+
+```bash
+conda env update -f environment.yml --prune
```

Comment:
So `--prune` would remove all packages *not in* the `environment.yml` file.

I don't know if we are affected/you're aware of this conda bug related to `--prune` @r4vi : https://github.com/conda/conda/issues/7279

Reply:
it's supposed to do that, and we might be affected by that bug but it's better to have the `--prune` in the docs so that when it's fixed everything will be :cool:

Reply:
agreed.